analyzers/src/SonarAnalyzer.Common/Rules/ShiftDynamicNotIntegerBase.cs

```diff
@@ -50,29 +51,21 @@ public abstract class ShiftDynamicNotIntegerBase<TExpressionSyntax> : SonarDiagn
             }
         }
 
-        private static string GetTypeNameForMessage(SyntaxNode expression, ITypeSymbol typeOfRight, SemanticModel semanticModel)
-        {
-            var constValue = semanticModel.GetConstantValue(expression);
-            return constValue.HasValue && constValue.Value == null
-                ? "null"
-                : typeOfRight.ToMinimalDisplayString(semanticModel, expression.SpanStart);
-        }
+        private static string GetTypeNameForMessage(SyntaxNode expression, ITypeSymbol typeOfRight, SemanticModel semanticModel) =>
+            semanticModel.GetConstantValue(expression) is { } constValue
```

Comment:
same as above, you can reduce this like this, meaning you don't even need the temp variable `constValue`.

```csharp
semanticModel.GetConstantValue(expression) is { HasValue: true, Value: null }
```

Reply:
That's great, and the readability improved too, thanks!